packages/unmock-core/src/generator-experimental.ts

```diff
@@ -0,0 +1,328 @@
+      const schema: Option<OpenAPIObject> = toSchema
+        .getOption(schemaRecord);
+      const operation: Option<Operation> = toSchema
+        .composeOptional(Optional.fromNullableProp<OpenAPIObject>()("paths"))
+        .composeIso(objectToArray())
+        .composeOptional(firstElementOptional())
+        .composeLens(valueLens())
+        .composeOptional(operationOptional)
+        .composeLens(valueLens())
+        .getOption(schemaRecord);
+      if (isNone(operation) || isNone(schema)) {
+        throw Error(`Cannot find a matcher for this request: ${JSON.stringify(req, null, 2)}`);
+      }
+      const codes: Array<(keyof Responses)> = Object.keys(operation.value.responses) as Array<keyof Responses>;
+      const code = codes[Math.floor(Math.random() * codes.length)];
+      const statusCode = code === "default" ? 500 : +code;
+      const definitions = schema.value.components && schema.value.components.schemas
+        ? Object.entries(schema.value.components.schemas)
+            .reduce((a, b) => ({...a, [b[0]]: isReference(b[1]) ? changeRef(b[1]) : changeRefs(b[1])}), {})
+        : {};
+      const headerSchema0 = Optional.fromNullableProp<Operation>()("parameters")
+        .composeTraversal(fromTraversable(array)())
+        .composePrism(new Prism(
+          s =>
+            isParameter(s)
+              ? some(s)
+              : isReference(s)
+              ? getParameterFromRef(schema.value, refName(s))
+              : none,
+          a => a,
+        ))
+        .composeOptional(parameterSchema(schema.value))
+        .composeGetter(identityGetter())
+        .getAll(operation.value);
+      const path1 = Optional.fromNullableProp<Operation>()("responses")
+      .composeOptional(Optional.fromNullableProp<Responses>()(code))
+      .composePrism(
+        new Prism(
+          s =>
+            isResponse(s)
+              ? some(s)
+              : isReference(s)
+              ? getResponseFromRef(schema.value, refName(s))
+              : none,
+          a => a,
+        ),
+      );
+      const stringHeader = (n: string, o: Option<Header>): Option<[string, Header]> =>
+        isNone(o) ? none : some([n, o.value]);
+      const headerSchema1 = path1
+        .composeOptional(Optional.fromNullableProp<Response>()("headers"))
+        .composeIso(objectToArray())
+        .composeTraversal(fromTraversable(array)())
+        .composePrism(new Prism<[string, Reference | Header], [string, Header]>(
+          a => isReference(a[1])
+            ? stringHeader(a[0], getHeaderFromRef(schema.value, refName(a[1])))
+            : some([a[0], a[1]]),
+          a => a,
+        ))
+        .composeIso(new Iso<[string, Header], Parameter>(
+          a => ({ ...a[1], name: a[0], in: "header"}),
+          a => [a.name, a],
+        ))
+        .composeOptional(parameterSchema(schema.value))
+        .composeGetter(identityGetter())
+        .getAll(operation.value);
+      const bodySchema = path1
```

Comment:
All these composed operations would in my opinion need some comments, they're quite abstract and hard to debug for anyone not familiar with monocle.

Reply:
Hm, yeah that is definitely a danger. The problem is that this is tightly coupled with `monocle-ts`. I'm not sure how to fix it, as it would essentially be documenting monocle. Maybe a comment on top of the file introducing people to monocle so that a maintainer felt comfortable with the main abstractions (Optional, Getter, Lens, Prism, Traversal, Iso).

Reply:
I'm not saying one should explain what monocle does, but I'm not sure even someone familiar with `monocle-ts` would see what this doing...:
```ts
      const headerSchema1 = path1
        .composeOptional(Optional.fromNullableProp<Response>()("headers"))
        .composeIso(objectToArray())
        .composeTraversal(fromTraversable(array)())
        .composePrism(new Prism<[string, Reference | Header], [string, Header]>(
          a => isReference(a[1])
            ? stringHeader(a[0], getHeaderFromRef(schema.value, refName(a[1])))
            : some([a[0], a[1]]),
          a => a,
        ))
        .composeIso(new Iso<[string, Header], Parameter>(
          a => ({ ...a[1], name: a[0], in: "header"}),
          a => [a.name, a],
        ))
        .composeOptional(parameterSchema(schema.value))
        .composeGetter(identityGetter())
        .getAll(operation.value);
```

Reply:
And while I agree with the notion that comments can be evil and the code should document itself, I think some help would maybe be in-place here and there 🙂

Reply:
Yeah, it's a really good point. The way I do it is hovering my mouse over in the IDE, seeing what comes in and out, and looking at the types of the schema to see how the optic is drilling down. What would help me is a suggestion for how to do the comments. I completely agree that they are necessary: if I saw this without an IDE, I would not touch this code, and even in an IDE where I get the types, I would be really intimidated.

Reply:
It's not worth commenting on every line, but maybe add a simple comment in every assignment (like they were isolated functions of their own) shortly explaining what it's doing and what's the example result, something like
```ts
/**
Extract header from...
Example result: ...
*/
const headerSchema1 = path1...
```
Also using more descriptive names instead of `a` in anonymous functions might help a bit. All this coming from someone with no experience in monocle-ts but just interested in reading the code from the perspective of unmock-js functionality.

Reply:
Changed, great idea!